### DISCUSSION TITLE
`Tag` and `BelongsTo::filterable()` does not respects Relatable Query and `relatableQueryUsing()`

### DISCUSSION THREAD

**Discussion by leith** [laravel/nova-issues#5097] [Ideas & Feature Requests] [answered]
- Laravel Version: 9.40.1
- Nova Version: 4.19.2
- PHP Version: 8.1.8
- Database Driver & Version: mysql (8.0.30)
- Operating System and Version: Windows 10 host, Ubuntu 20.04 virtual guest (Homestead)
- Browser type and version: Chrome 107.0
- Reproduction Repository: -

### Description:
List of `<select>` options for a `BelongsTo` field when using `filterable()` does not respect `relatableQueryUsing()`

### Detailed steps to reproduce the issue on a fresh Nova installation:

Two database tables:
* `categories` : `id` (id), `level` (tinyint), `name` (string)
* `products` : `id` (id), `category_1_id` (foreignId), `category_2_id` (foreignId), `name` (string)

A product `belongsTo` a primary category and a secondary category, determined by the `level` attribute in the categories table.

`App/Nova/Product`:
```php
    public function fields(NovaRequest $request)
    {
        return [
            BelongsTo::make('Primary Category', 'primary_category', Category::class)
                ->relatableQueryUsing(function (NovaRequest $request, Builder $query) {
                    $query->where('level', 1);
                })
                ->filterable(),

            // ...

        ];
    }
```
When viewing the `Product` resource list, the filter is created, but it contains **all** possible categories, not just those limited by the `relatableQueryUsing()` callback.

While `filterable()` has a callback option for modifying a query, that query is for the results of the filter, not the options presented as a part of the filter.

**Comment by crynobone** [comment]
Marking this as feature request since `RelatableQuery` <https://nova.laravel.com/docs/4.0/resources/authorization.html#relatable-filtering> isn't being used as well.

**Reply by leith** [reply]
As far as I'm aware, it's not possible to use `RelatableQuery` in this format, since both primary and secondary category fields use the same model and it would override the query for both fields?

**Reply by crynobone** [reply]
Yes, `relatableQueryUsing` reflect `RelatableQuery` feature, and `filterable()` doesn't implements that.

**Reply by puzzledmonkey** [reply]
@crynobone is it possible to determine whether it's from a filter or a normal query by way of the NovaRequest $request if using Dynamic Relatable Methods as per https://nova.laravel.com/docs/4.0/resources/authorization.html#relatable-filtering ?

**Reply by crynobone** [reply]
I feel that's irrelevant to the issue. Both features query the API using the same controller and it not utilising Relatable Query, that's all.

**Comment by MannikJ** [comment]
What is the current state of filterable on a BelongsTo relation? Although this feature seems very cool, it is also unsuable in the way it behaves right now if I'm not missing anything. 

I just investigated that our server crashes where caused because Nova was loading ALL damn records for each filter on a BelongsTo relation. I have not yet thought about relatableQuery stuff, but it's at least related.
I assumed that appending the searchable attribute should be enough to tell Nova that this relation is not qualified for being loaded in its entirety. 

I found that the vue filter component when initialized makes a request like this:
https://my-app.local/nova-api/my-resource/search?current=&first=&search=&withTrashed=false

**Comment by Aniruddh-J** [comment]
Do we have any update on this? Or is there any workaround to make the Tag field respect a relatable query?

**Comment by crynobone** [comment, marked as answer]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
